test/common/grpc/grpc_web_filter_test.cc

```diff
@@ -39,6 +39,37 @@ class GrpcWebFilterTest : public testing::Test {
   NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
 };
 
+TEST_F(GrpcWebFilterTest, SupportedContentTypes) {
+  Http::TestHeaderMapImpl request_headers;
+  // Check "content-type:application/grpc-web" is supported.
+  request_headers.addViaCopy(Http::Headers::get().ContentType,
+                             Http::Headers::get().ContentTypeValues.GrpcWeb);
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
+  EXPECT_EQ(Http::Headers::get().ContentTypeValues.Grpc,
+            request_headers.ContentType()->value().c_str());
+
+  // Check "content-type:application/grpc-web+proto" is supported.
+  request_headers.addViaCopy(Http::Headers::get().ContentType,
+                             Http::Headers::get().ContentTypeValues.GrpcWeb_Proto);
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
+  EXPECT_EQ(Http::Headers::get().ContentTypeValues.Grpc,
+            request_headers.ContentType()->value().c_str());
+
+  // Check "content-type:application/grpc-web-text" is supported.
+  request_headers.addViaCopy(Http::Headers::get().ContentType,
+                             Http::Headers::get().ContentTypeValues.GrpcWebText);
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
+  EXPECT_EQ(Http::Headers::get().ContentTypeValues.Grpc,
+            request_headers.ContentType()->value().c_str());
+
+  // Check "content-type:application/grpc-web-text+proto" is supported.
+  request_headers.addViaCopy(Http::Headers::get().ContentType,
+                             Http::Headers::get().ContentTypeValues.GrpcWebText_Proto);
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
```

Comment:
You could rewrite this as a `for` loop over an array literal for `ContentTypeValues` to reduce the boiler plate (it could be like 7 LoC). Also, should we have a negative test for when a protocol isn't supported?